packages/context/test/unit/resolver.test.ts

```diff
@@ -320,3 +326,90 @@ function customAsyncDecorator(def: Object) {
     return f + ':' + b;
   });
 }
+
+describe('method injection', () => {
+  let ctx: Context;
+
+  before(function() {
+    ctx = new Context();
+    ctx.bind('foo').to('FOO');
+    ctx.bind('bar').to('BAR');
+  });
+
+  it('can resolve method arguments', () => {
+    class TestClass {
+      test(@inject('foo') foo: string) {
+        return `hello ${foo}`;
+      }
+    }
+
+    const t = invokeMethod(TestClass.prototype, 'test', ctx);
+    expect(t).to.eql('hello FOO');
+  });
+
+  it('can resolve method arguments with instance', () => {
+    class TestClass {
+      bar: string;
+
+      test(@inject('foo') foo: string) {
+        this.bar = foo;
+        return `hello ${foo}`;
```

Comment:
Since handling of return values is already covered by the previous test, we should change `test` into a function returning `void` in order keep this test narrowly focused and make it easier for readers to understand what is actually being tested.

```ts
test(@inject('foo') foo: string) {
  this.bar = foo;
}
```